src/main/java/uk/nhs/digital/nhsconnect/nhais/service/edifact_to_fhir/DrugsMarkerExtensionPatchTransactionMapper.java

```diff
@@ -0,0 +1,42 @@
+package uk.nhs.digital.nhsconnect.nhais.service.edifact_to_fhir;
+
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.Transaction;
+import uk.nhs.digital.nhsconnect.nhais.model.fhir.DrugsMarkerExtension;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentBooleanExtension;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatch;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentValue;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class DrugsMarkerExtensionPatchTransactionMapper implements PatchTransactionMapper {
+
+    @Override
+    public AmendmentPatch map(Transaction transaction) {
+        var drugMarker = transaction.getDrugsMarker();
+        if (drugMarker.isPresent()) {
+            var drugMarketBoolean = drugMarker.get().getDrugsMarker();
+            return createAmmendmentExtensionPatch(drugMarketBoolean);
+        } else {
+            return null;
+        }
+    }
+
+    private AmendmentPatch createAmmendmentExtensionPatch(boolean element) {
+        var path = "/extension/0";
+
+        AmendmentPatch amendmentPatch = new AmendmentPatch();
+        if (!element){
+            amendmentPatch.setOp(AmendmentPatchOperation.REMOVE);
+            amendmentPatch.setPath(path);
+            amendmentPatch.setValue(new AmendmentBooleanExtension(DrugsMarkerExtension.URL, "false"));
+        } else {
+            amendmentPatch.setOp(AmendmentPatchOperation.REPLACE);
+            amendmentPatch.setPath(path);
+            amendmentPatch.setValue(new AmendmentBooleanExtension(DrugsMarkerExtension.URL, "true"));
```

Comment:
Those parts should be extracted to methods for readability and maintaining single level of abstraction. See my example implementation in another comment

Reply:
Or you could just use constructors
````java
private AmendmentPatch createAmendmentExtensionPatch(boolean drugMarker) {
    var path = "/extension/0";
    if (drugMarker){
        return new AmendmentPatch(AmendmentPatchOperation.REPLACE, path, new AmendmentBooleanExtension(DrugsMarkerExtension.URL, "true"));
    }
    return new AmendmentPatchRemoval(path, new AmendmentBooleanExtension(DrugsMarkerExtension.URL, "false"));
}
````